PythonAPI/examples/autoware_demo.py

```diff
@@ -174,7 +186,7 @@ def generate_gnss_blueprint(blueprint_library):
     return blueprint
 
 
-def spawn_sensors(world, base_link, ego):
+def spawn_sensors(world, base_link, ego, additional_args=None):
```

Comment:
```suggestion
def spawn_sensors(world, base_link, ego, additional_args):
```
I do not think this is a good idea, as you can omit this argument and then python will raise an error - it is **never** checked whether the `additional_args` is `None`.

This suggestion is the fastest and easiest fix, but this can also be fixed with some dedicated config/settings class that will always have a method or data member set with some default value.

> [!NOTE]
> This default value `None` should in my opinion be removed from all other functions that pass it around, I am just only mentioning it here.

Reply:
I agree. I really dislike this solution with passing arguments anyway, but this seems to be the fastest solution.
I think a general refactor of `autoware_demo.py` would be needed - but in a separate PR.